File: android/ffmpeg-kit-android-lib/src/main/java/com/lgcns/ffmpegkit/AsyncFFprobeExecuteTask.java


package com.lgcns.ffmpegkit;

import com.arthenica.smartexception.java.Exceptions;

/**
 * <p>Executes an FFprobe session asynchronously.
 */
public class AsyncFFprobeExecuteTask implements Runnable {
    private final FFprobeSession ffprobeSession;
    private final FFprobeSessionCompleteCallback completeCallback;

    public AsyncFFprobeExecuteTask(final FFprobeSession ffprobeSession) {
        this.ffprobeSession = ffprobeSession;
        this.completeCallback = ffprobeSession.getCompleteCallback();
    }

    @Override
    public void run() {
        FFmpegKitConfig.ffprobeExecute(ffprobeSession);

        if (completeCallback != null) {
            try {
                // NOTIFY SESSION CALLBACK DEFINED
                completeCallback.apply(ffprobeSession);
            } catch (final Exception e) {
                android.util.Log.e(FFmpegKitConfig.TAG, String.format("Exception thrown inside session complete callback.%s", Exceptions.getStackTraceString(e)));
            }
        }

        final FFprobeSessionCompleteCallback globalFFprobeSessionCompleteCallback = FFmpegKitConfig.getFFprobeSessionCompleteCallback();
        if (globalFFprobeSessionCompleteCallback != null) {
            try {
                // NOTIFY GLOBAL CALLBACK DEFINED
                globalFFprobeSessionCompleteCallback.apply(ffprobeSession);
            } catch (final Exception e) {
                android.util.Log.e(FFmpegKitConfig.TAG, String.format("Exception thrown inside global complete callback.%s", Exceptions.getStackTraceString(e)));
            }
        }
    }

}

File: android/ffmpeg-kit-android-lib/src/main/java/com/lgcns/ffmpegkit/AsyncFFprobeExecuteTask.kt
package com.lgcns.ffmpegkit

import android.util.Log

/**
 *
 * Executes an FFprobe session asynchronously.
 */
class AsyncFFprobeExecuteTask(private val ffprobeSession: FFprobeSession) : Runnable {
    private val completeCallback: FFprobeSessionCompleteCallback? = ffprobeSession.completeCallback

    override fun run() {
        FFmpegKitConfig.ffprobeExecute(ffprobeSession)
        if (completeCallback != null) {
            try {
                // NOTIFY SESSION CALLBACK DEFINED
                completeCallback.apply(ffprobeSession)
            } catch (e: Exception) {
                Log.e(
                    FFmpegKitConfig.TAG,
                    String.format("Exception thrown inside session complete callback.%s", e.message)
                )
            }
        }
        val globalFFprobeSessionCompleteCallback =
            FFmpegKitConfig.getFFprobeSessionCompleteCallback()
        if (globalFFprobeSessionCompleteCallback != null) {
            try {
                // NOTIFY GLOBAL CALLBACK DEFINED
                globalFFprobeSessionCompleteCallback.apply(ffprobeSession)
            } catch (e: Exception) {
                Log.e(
                    FFmpegKitConfig.TAG,
                    String.format("Exception thrown inside global complete callback.%s", e.message)
                )
            }
        }
    }
}